crates/biome_configuration/src/graphql.rs

```diff
@@ -55,7 +55,12 @@ pub struct GraphqlFormatter {
         argument("double|single"),
         optional
     ))]
-    pub quote_style: QuoteStyle,
```

Comment:
Why do we have some options as Option<T> and others as <T>?

Reply:
We recently refactored some of the code (JSON and CSS especialy) to use `PartialXXXConfiguration` when mapping the options to `LanguageSettings`. In a `Partial`-prefixed struct every option is an `Option<T>`. But this is yet to be done for the Js language, see #3297, and maybe `GraphQL` as it is newly added. Until the refactoring is completed, using `Option<T>` should be a more safe choice.

And honestly, I don't know if there're cases where we shouldn't use `Partial`-prefixed structs, and am not very clear why we instroduce the `partial` macro if every option is already wrapped in an `Option`. I'm also not clear why we use the `partial` macros for primitive values.

Reply:
@arend would you mind giving some background here?

Reply:
@arendjr would you mind giving us some background here? I admit that I also lack the knowledge, so maybe we should have it somewhere

Reply:
So the goal of the `Partial` types is to enable merging of configs. With fields that are not wrapped in `Option` you may not know if a field is set and just happened to be set to the default value or if it was not set at all. The distinction matters because when merging one should override values in the config it is being merged into, while the other shouldn’t.

Configurations that are already merged don’t need to use `Partial` types anymore, and their fields generally don’t need to be wrapped in `Option` as long as there’s a sensible default. Of course, sometimes it may still make semantic sense to use an `Option` wrapper for given field, and I think in such a case the derived `Partial` type won’t wrap the `Option` again.

For primitive types, they indeed don’t need to have a `Partial` version, but they still implement the `Mergeable` trait (I’m not entirely sure about the trait name anymore) so that the derive macro can treat all types consistently.

> We recently refactored some of the code (JSON and CSS especialy) to use PartialXXXConfiguration when mapping the options to LanguageSettings

What was the goal of this? When creating the `Partial` types the idea was that `XXXConfiguration` could define all the default values and would be the result of the merge process, so `PartialXXXConfiguration` would only be used before merging. I suppose you could indeed go directly from `Partial` types to `Settings` but you’ll need a different place to define the defaults I guess. I’m not very partial (pun intended) to which approach we use, as long as we’re aligned we go the right direction.

Reply:
> When creating the `Partial` types the idea was that `XXXConfiguration` could define all the default values and would be the result of the merge process, so `PartialXXXConfiguration` would only be used before merging. I suppose you could indeed go directly from `Partial` types to `Settings` but you’ll need a different place to define the defaults I guess. I’m not very partial (pun intended) to which approach we use, as long as we’re aligned we go the right direction.

In our current infra structure we map `Configuration` (direct input) to `Settings` (our internal workspace struct), then to `Options` (parse/format/lint functions). The reason we need the `Configuration` to be `partial` is that the overrides are calculated in the `Settings` phase, so we need the `None` type to know whether an option can override the existing one. So in #3272 and #3273 we change all fields in the `Settings` to `Option<T>`, this will also require us to use the `PartialXXXConfiguration` for mapping options to `Settings`.

We only keep concrete/merged fields in the `Options` structs.

But I still think we should keep the defaults in the `Configuration` structs though, and wrap all the primitives with custom types so we can provide custom defaults. Just a half-baked idea.

Reply:
Oh, I think it's inevitable to move defaults to the `Settings` struct...

Reply:
Yeah, I guess if we put the defaults in `Settings` and make sure all the `Configuration` fields are all `Option`s already we can drop the whole `Partial` concept since it becomes kinda redundant at that point. Think that might actually simplify things.

Reply:
Weren't Partial meant to handle merging from `extends`?

Reply:
Yes, but we also still have a separate trait for that IIRC. The `Partial` types are just to enforce that all fields are wrapped in `Option`. But if we want to forward everything as an `Option` to the `Settings` anyway, I think it might be sufficient to document it like that and we may not need the separate macro anymore.

Reply:
Just checked, it’s the `Merge` derive macro that does the actual merging for the `extends` feature.

Reply:
If we ensure `Configuration` fields are all `Option`s we can just drop the `partial` macro and `extends` will still work.

Reply:
If we really want enforcement maybe we could even let the `Merge` macro raise an error if a field isn’t an `Option`, with a message explaining why. But then at least we don’t need separate types for it anymore.

Reply:
Just noticed we use `Configuration` for the `rage` command. So if we change every field to an `Option<T>`, we need an approach to retrieve defaults from the `Configuration` structs. 

Priorly said, we also need defaults when resolving `Options` from `LanguageSettings`. I wonder if there's a recommended way to keep a single source of truth for all the defaults. So they can be used in both places. Maybe a `DefaultXXX` struct that holds the default values and we can implement `From` traits for the consumers?